sendgrid.js

```diff
@@ -17,16 +17,28 @@ function marshallEvent (event) {
   const timestamp = mapTimestamp(event.timestamp)
   const mail = marshallMailObject(event, timestamp)
 
+  const notification = {
+    Type: 'Notification',
+    Timestamp: new Date().toISOString()
+  }
+
   switch (event.event) {
     case EVENTS.BOUNCE:
     case EVENTS.DROPPED:
-      return { mail, ...marshallBounceEvent(event, timestamp) }
-
+      return { 
+        ...notification, 
+        Message: JSON.stringify({ mail, ...marshallBounceEvent(event, timestamp) })
+      }
     case EVENTS.DELIVERED:
-      return { mail, ...marshallDeliveryEvent(event, timestamp) }
-
+      return { 
+        ...notification, 
+        Message: JSON.stringify({ mail, ...marshallDeliveryEvent(event, timestamp) })
+      }
     case EVENTS.SPAM:
-      return { mail, ...marshallComplaintEvent(event, timestamp) }
+      return { 
+        ...notification, 
+        Message: JSON.stringify({ mail, ...marshallComplaintEvent(event, timestamp) })
+      }
```

Comment:
So following on from other comment, I mean eliminate the duplication of doing this stuff out here and push it down one level into `index.js` instead.

Reply:
Even when I send this down to the `index.js` level, when asseting we check for the arguments in `sqs.push` and that is always going to be the stringified JSON. I could just assert against the stringified version... I don't know if that is better than parsing it again. What do you think @philbooth ?